layout.js

```diff
@@ -1,23 +1,51 @@
+/**
+ * 
+ * Updated Aug 9-10, 2017 by Rhett Anderson
+ * 
+ * The spriter is a naive packer. It sorts images by size and then plops them down into a spritesheet. It's
+ * very willing to make a long horizontal line until it crosses the 2048 pixel boundary. This tends to leave
+ * a lot of unused texture memory.
+ * 
+ * Rather than rewrite the naive packer, I run it at different maximum widths (2048, 1024, 512) and compare
+ * the outcomes. I choose the one with the smallest number of pixels (area).
+ * 
+ * This is a pretty brute-force solution to improve the texture memory situation. A better next step than
+ * improving this would be to use a proven solution like texturePackerPro.
+ * 
+ */
+
+
 var Spritesheet = require('./Spritesheet');
 
 var DEFAULT_MAX_SIZE = 2048;
+var DEFAULT_MIN_SIZE = 512;
 var DEFAULT_PADDING = 2;
 
 /**
  * @returns {Spritesheet[]}
  */
+
 module.exports = function (images, opts) {
+  var bestArea = -1;
+  var bestNumSheets = 1000;
+  var bestX = DEFAULT_MAX_SIZE;
+
   if (!opts) { opts = {}; }
-  var maxSize = opts.maxSize !== undefined ? opts.maxSize : DEFAULT_MAX_SIZE;
+  var maxSizeX = opts.maxSize !== undefined ? opts.maxSize : DEFAULT_MAX_SIZE;
```

Comment:
This is an example comment for how I could say something meaningful but all I've done is demonstrate comments.